### PR TITLE
Enh/v1/code packaging

### LINKED ISSUE #11: modularize and Package the code
Refactor and reformat the code as needed, also package the code as library installation.

### PR DESCRIPTION
**Pull request by pushvinderrohtagi**
1.Modularizing and packaging  the code
2.creating unit and functional tests

### CODE REVIEW

File: .readthedocs.yml
Comment (ankitamalarya) [inline .readthedocs.yml:1]: Is this required here?

File: bin/README.md
Comment (ankitamalarya) [inline bin/README.md:1]: bin folder can be added to gitignore

File: deploy/requirements.txt
Comment (ankitamalarya) [inline deploy/requirements.txt:1]: add package versions

File: install_housepred.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    echo "mle-dev env exists already. house prediction package and dependencies will be installed in the existing env"
+    conda activate mle-dev
+else
+    echo "Creating a new env mle-dev where house prediction package and dependencies will be installed"
+    conda create -n mle-dev python=3.8 -y
+    conda activate mle-dev
+fi
+
+# Conda install HDBSCAN package as pip install hdbscan has gcc incompatibility issues
+conda install -c conda-forge hdbscan==0.8.29 -y
```
Comment (ankitamalarya) [inline install_housepred.sh]: do we need hdbscan?

File: env.yml
Comment (ankitamalarya) [inline env.yml:1]: where is this file used?
I see both requirements.txt and env yaml, Which one is to be used?

File: tests/functional_tests/test_data_prep.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import pytest
+
+from src.data_prep import income_cat_proportions
```
Comment (ankitamalarya) [inline tests/functional_tests/test_data_prep.py]: imports should be from package name
Example if package name is housepred do from housepred.data_prep import xyz

File: .readthedocs.yml
Comment (pushvinderrohtagi) [reply .readthedocs.yml:1]: Not required, deleted

File: bin/README.md
Comment (pushvinderrohtagi) [reply bin/README.md:1]: added

File: deploy/requirements.txt
Comment (pushvinderrohtagi) [reply deploy/requirements.txt:1]: done

File: install_housepred.sh (hunk 1 above)
Comment (pushvinderrohtagi) [reply install_housepred.sh]: no we dont, removed

File: env.yml
Comment (pushvinderrohtagi) [reply env.yml:1]: this was mentioned in the instructions to create a env.yml file for development

File: tests/functional_tests/test_data_prep.py (hunk 2 above)
Comment (pushvinderrohtagi) [reply tests/functional_tests/test_data_prep.py]: done

**Review by ankitamalarya — APPROVED** [review]

**Pull request merged**